### PR TITLE
Doc/ar 34/en tant quexaminateur je souhaite avoir un readme md decrivant lapi du server

### PR DESCRIPTION
**Pull request by Vivelis**
* **Checklist before requesting a review**
- [x] The commits messages follows our guidelines
- [x] I have performed a self-review of my code
- [x] I have documented my code
- [ ] Do we need to implement unit tests?

* **In what state is your pull request?**
- [x] Ready to merge / Waiting a reviwer to see my work.
- [ ] Work In Progress (WIP) / My work is not finish but i want daily reviews. (Draft)
- [ ] CI Review / I want to see what the CI thinks of my work.
- [ ] Need feedback / Just to have feedback on what i produced. (May not be merged)

* **What kind of changes does this PR introduce?** (You can choose multiple)
- [ ] Bug fix
- [ ] Feature request
- [x] New / Updated documentation
- [ ] Testing CI ( Make the pull request in draft mode)

* **Describe your changes**
- add user documentation
- add api documentation
- fix documentation

* **Issue ticket number and link**
[AR-34](https://bravmm.atlassian.net/browse/AR-34)

[AR-34]: https://bravmm.atlassian.net/browse/AR-34?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by CodeRabbit

- **Documentation**
    - Updated project details and installation instructions in README.
    - Enhanced developer documentation with framework changes and new sections for API and Spotify integration.
    - Added a database diagram to the DB documentation.
    - Updated front-end documentation to reflect the shift to NextJs.
    - Refined mobile application documentation formatting.
    - Expanded user documentation with new instructions and visual aids.

- **New Features**
    - Implemented a new home page section for user documentation.

- **Refactor**
    - Modified user information decryption process for enhanced security.

- **Style**
    - Integrated `TextSection` component for dynamic content in registration page.

- **Tests**
    - Added cookie-based authentication checks in services page logic.

### CODE REVIEW

File: front/app/services/page.tsx [hunk 1]
```diff
@@ -121,6 +122,37 @@ export default function Services() {
...
+        if (response.status !== 200) {
+          console.log("Not logged in");
+          router.push("/");
+        }
+      });
+    } catch (error) {
+      console.error("Error:", error);
+    }
+  }
+  , []);
```
[the remark this answers is not in the document]
Comment (Vivelis) [reply front/app/services/page.tsx:154]: done in another branch

File: README.md [hunk 2]
```diff
@@ -1,48 +1,72 @@
...
+
+If you want to contribute to the project, you should start by reading the [developer documentation](docs/developer_documentation/index.md).
 
-Access the API via the address: http://localhost:8080
+### New user
 
-Access the mobile application via expo (auto search)
+If you want to learn how to use the project, you should start by reading the [user documentation](docs/user_documentation/index.md).
 
 ## Database Diagram
```
[the remark this answers is not in the document]
Comment (Vivelis) [reply README.md:72]: can't do that

[the remark this answers is not in the document]
Comment (Vivelis) [reply README.md:72]: can't do that

**Review by Beafowl-Pull — DISMISSED** [review]
RAS

**Review by Beafowl-Pull — DISMISSED** [review]
RAS

**Review by romainpanno — APPROVED** [review]

**Review by Beafowl-Pull — APPROVED** [review]
RAS

**Review by KitetsuK — DISMISSED** [review]

**Review by Beafowl-Pull — APPROVED** [review]
RAS

**Review by KitetsuK — APPROVED** [review]

**Review by romainpanno — DISMISSED** [review]

**Review by romainpanno — DISMISSED** [review]

**Review by Beafowl-Pull — DISMISSED** [review]

**Review by Beafowl-Pull — APPROVED** [review]

**Review by romainpanno — APPROVED** [review]

**Review by romainpanno — APPROVED** [review]

**Review by KitetsuK — DISMISSED** [review]

**Review by Beafowl-Pull — DISMISSED** [review]

**Pull request merged**